### DISCUSSION TITLE
Request: More support for subdomains and reverse proxying

### DISCUSSION THREAD

**Discussion by nfowler** [Syncplay/syncplay#672] [Ideas]
I'm trying to use Caddy as a reverse proxy, allowing users to connect to my Syncplay server via a subdomain such as syncplay.example.com.

I would like to do this with a configuration such as:
 ```
syncplay.example.com {
    reverse_proxy 000.000.000.000:XXXX
}
```
Wherein, to explain, a request sent to syncplay.example.com is sent to my server's public IP, at the port servicing Syncplay.

Unfortunately, the Syncplay client aggressively appends a port to your server address. If no port is detected, it seems to append the default 8999 port. As a result, trying to enter syncplay.example.com as my server address will fail, as it instead tries to connect to syncplay.example.com:8999, which will error out even if the port for the server is in fact 8999.

The only solution I've found is to change the Caddy configuration to something more like:
```
syncplay.example.com {
    reverse_proxy 000.000.000.000
}
```
And simply requiring users to append your port or, if you're using the default port, simply trust the client to do so. I'm not very pleased with this solution.

Is there a way to, for example, disable automatic port appending if a subdomain is detected?

**Comment by Et0h** [comment]
Syncplay does not include any explicit support for proxies or reverse proxies, it is designed for direct connection to a client-specified server on a client-specified port.

See https://github.com/Syncplay/syncplay/issues/346 (specifically EtiennePerot's comment on 24th September 2020 and tacerus's comment on 31st July 2022) for details on using Syncplay with (reverse?) proxies.

**Comment by albertosottile** [comment]
@nfowler Just out of curiosity, what happens if you try to use Syncplay to connect to `syncplay.example.com:80`?

**Comment by scientificworld** [comment]
You can try [caddy-l4](https://github.com/mholt/caddy-l4) (and use tacerus's script in front) for proxying.

Here is my configuration:
```json
{
	"apps": {
		"layer4": {
			"servers": {
				"example": {
					"listen": ["0.0.0.0:8998"],
					"routes": [
						{
							"match": [
								{
									"tls": {
										"sni": ["syncplay.example.com"]
									}
								}
							],
							"handle": [
								{
									"handler": "tls"
								},
								{
									"handler": "proxy",
									"upstreams": [
										{"dial": ["localhost:8999"]}
									]
								}
							]
						}
					]
				}
			}
		},
		"tls": {
			"certificates": {
				"automate": ["syncplay.example.com"]
			},
			"automation": {
				"policies": [
					{
						"issuers": [{"module": "zerossl"}]
					}
				]
			}
		}
	}
}
```

**Comment by port19x** [comment]
> @nfowler Just out of curiosity, what happens if you try to use Syncplay to connect to `syncplay.example.com:80`?

This

![2023-09-19_15-01_1](https://github.com/Syncplay/syncplay/assets/82055622/8b0f4613-05e7-4448-9005-20d7fa19a302)
